### ISSUE TITLE
client: handle USER_MIGRATE

### ISSUE THREAD

**Issue by daodao2007** [gotd/td#100]
Incomplete code implementation
When I tested my code, I passed the proxy connection to the server, and I always received the 303 message at the rpcDo function.

According to my reading of the code https://github.com/Lonami/grammers,
https://github.com/Lonami/grammers/blob/master/lib/grammers-client/src/client/auth.rs
bot_sign_in function

```
Err(InvocationError::Rpc(RpcError { name, value, .. })) if name == "USER_MIGRATE" => {
                self.config.session.auth_key = None;
                self.sender = connect_sender(value.unwrap() as i32, &mut self.config).await?;
                self.invoke(&request).await?
            }
```

When the 303 message is received, the rpc connection should be switched to the parameter number dc server in the 303 message.
There is no such operation in the current code. Since I am not familiar with the code you wrote, I cannot complete this part of the code yet, so please add it yourself.

The echo example has worked, thank you for your contribution.

**Comment by ernado** [comment]
Hey, automatic connection migration is not currently implemented, but is on roadmap.

**Comment by ernado** [comment]
Released in v0.17
